### ISSUE TITLE
Sort out what can change when doing a restart

### ISSUE THREAD

**Issue by rjleveque** [clawpack/geoclaw#435]
As mentioned in https://github.com/clawpack/geoclaw/pull/434, adding support for Lagrangian gauges (particle tracking) required doing some things in a different order for a restart.

More generally we should figure out what things are allowed to change when doing a restart (e.g. one can add new gauges but not remove old ones, can add additional levels but not have fewer) and make sure things are called in the right order between what is read in from the checkpoint file and what is set by reading a `.data` file, and better document all this.

**Comment by mjberger** [comment]
I was looking at the restart code to see why the number of levels was not allowed to decrease, only increase.

Turns out there is an old comment line that says that, but the code actually allowed both.

I guess the comment should be updated.  :)

 — Marsha

**Comment by rjleveque** [comment]
Yes, I guess we should update the comment, also in amrclaw.  So I guess we just throw away the finer level data if restarting with fewer levels.  Good to know.
